### DISCUSSION TITLE
Build an older version of INAV (Version 1.73)

### DISCUSSION THREAD

**Discussion by julioghigi** [iNavFlight/inav#7029] [Q&A] [answered]
Hello guys!

I am trying to build an older version of INAV, in order to remap some pins and got an older FC working with an flying wing.

I have an "old" SPRacing F3 laying around, and a cousin of mine is starting on the hobby and wann fly his new flying wing, but he still only have those somple 4 channels radio, without any kind of mix and whatever, so I wanna install this ols SPRacing F3 on his flying wing in order to read the PWM inputs from his rx and makes any kind of mix a Flying wing needs, and give some stabilization also, since he is starting in the hobby.

But as all we know, INAV took off the capability of read PWM input (and I tottaly understand and agree), so I figured out that the INAV 1.73 works really fine to read the PWM inputs so we can keep going with his Flying wing. BUT this FC has some pins broken and I can`t get it working with PWM input 1 and 2, but normally with input 3, 4, 5, 6, 7. So let's do some remaping on the pins!

I did a Git repository on my PC, got everything working here (I guess) but with the actual INAV files, so I put the command `$ git checkout 974440d` wich looks like me send all my files to the 1.73 version, GOOD!

But I am still not able to build it, the actual WIKI commands only works with the newest versions, do someone know wich command I need to do to build? I still didn't change any file at all, still trying to build it as clean as possible, and making it work, start doing some changes.

I will be extremely thankful to whatever who helps me! I need this ASAP since I wanna fly tomorrow with him.

Sorry for my bad english!

Thanks in advance!

**Comment by stronnag** [comment]
If you check out an old commit, you will get the relevant documentation in the `docs/development/` directory.

**Comment by julioghigi** [comment, marked as answer]
I get on the documentation the following code to build `make TARGET=SPRACINGF3`, but was still not building because it was acousing no Ruby installed (I did installed and was working fine on a normal command prompt)

I can build it now, but it need to be done by the Deprecated way on windows, in this link: https://github.com/iNavFlight/inav/blob/master/docs/development/Building%20in%20Windows%20light.md

and after all installed, is it still necessary to edit `shF4.cmd` file in order to correct the path to ruby (lost hours trying to figure out why it wasn't founding Ruby.

And after being beat a lot remapping the pins and failing, I got it done! It is working fine and soon will be flying!

Thanks a lot, this issue (not tottaly a issue) can be closed.
